packages/shell-api/src/shard.ts

```diff
@@ -811,4 +811,26 @@ export default class Shard extends ShellApiWithMongoClass {
       abortUnshardCollection: ns,
     });
   }
+
+  @serverVersions(['8.0.0', ServerVersions.latest])
+  @apiVersions([])
+  @returnsPromise
+  async isConfigShardEnabled(): Promise<Document> {
+    this._emitShardApiCall('isConfigShardEnabled', {});
+    await getConfigDB(this._database);
+
+    const shards = (await this._database.adminCommand({ listShards: 1 }))[
+      'shards'
+    ] as Document[] | undefined;
```

Comment:
Surprised the linter doesn't complain, we'd probably want roughly this, right?

```suggestion
    const shards = (await this._database.adminCommand({ listShards: 1 })).shards as Document[] | undefined;
```